### PR TITLE
Removes unused prop types; ESLint's preventing them.

### PR DESCRIPTION
**Pull request by skiplist**
#### Summary

Removes unused prop type validation; set ESLint to error when present.

#### Ticket Link

n/a

### CODE REVIEW

**Comment by jespino** [comment]
I want to review it in more detail, but for sure I love it :)

File: components/multiselect/multiselect_list.jsx [hunk 1]
```diff
@@ -177,9 +177,6 @@ MultiSelectList.defaultProps = {
 MultiSelectList.propTypes = {
     options: PropTypes.arrayOf(PropTypes.object),
     optionRenderer: PropTypes.func,
-    page: PropTypes.number,
-    perPage: PropTypes.number,
```
Comment (mgdelacroix) [inline components/multiselect/multiselect_list.jsx]: Here we can remove `perPage` from the `defaultProps` and stop passing `page`, `perPage` and `onPageChange` from the `multiselect.jsx` file.

File: components/new_channel_modal/new_channel_modal.jsx [hunk 2]
```diff
@@ -25,11 +25,6 @@ export default class NewChannelModal extends React.PureComponent {
          */
         show: PropTypes.bool.isRequired,
 
-        /**
-         * Id of the active team
-         */
-        currentTeamId: PropTypes.string.isRequired,
```
Comment (mgdelacroix) [inline components/new_channel_modal/new_channel_modal.jsx]: Here we can stop passing `currentTeamId` from the index

File: components/post_view/post/post.jsx [hunk 3]
```diff
@@ -48,11 +48,6 @@ export default class Post extends React.PureComponent {
          */
         consecutivePostByUser: PropTypes.bool,
 
-        /**
-         * Set if the previous post is a comment
-         */
-        previousPostIsComment: PropTypes.bool,
```
Comment (mgdelacroix) [inline components/post_view/post/post.jsx]: Here `previousPostIsComment` is in use as part (at least) of the `hasSameRoot` function

File: components/post_view/post_body/post_body.jsx [hunk 4]
```diff
@@ -67,11 +62,6 @@ export default class PostBody extends React.PureComponent {
          */
         isEmbedVisible: PropTypes.bool,
 
-        /**
-         * Whether or not the post username can be overridden.
-         */
-        enablePostUsernameOverride: PropTypes.bool.isRequired,
```
Comment (mgdelacroix) [inline components/post_view/post_body/post_body.jsx]: Here we can stop passing both `parentPostUser` and `enablePostUsernameOverride` from the index

File: components/post_view/post_message_view/post_message_view.jsx [hunk 5]
```diff
@@ -41,16 +41,6 @@ export default class PostMessageView extends React.PureComponent {
...
 
-        /**
-         * Whether or not the RHS is visible
-         */
-        isRHSOpen: PropTypes.bool,
-
-        /**
-         * Whether or not the RHS is expanded
-         */
-        isRHSExpanded: PropTypes.bool,
```
Comment (mgdelacroix) [inline components/post_view/post_message_view/post_message_view.jsx]: We can stop passing `isRHSOpen` and `isRHSExpanded` from the index

File: components/search_results/search_results.jsx [hunk 6]
```diff
@@ -47,7 +47,6 @@ export default class SearchResults extends React.PureComponent {
     static propTypes = {
         results: PropTypes.array,
         matches: PropTypes.object,
-        currentUser: PropTypes.object,
```
Comment (mgdelacroix) [inline components/search_results/search_results.jsx]: We can stop passing `currentUser` from the index and remove it from the `defaultProps`

File: components/single_image_view/single_image_view.jsx [hunk 7]
```diff
@@ -20,7 +20,6 @@ export default class SingleImageView extends React.PureComponent {
     static propTypes = {
         postId: PropTypes.string.isRequired,
         fileInfo: PropTypes.object.isRequired,
-        isRhsOpen: PropTypes.bool.isRequired,
```
Comment (mgdelacroix) [inline components/single_image_view/single_image_view.jsx]: We can stop passing `isRhsOpen` from the index

File: components/user_settings/display/user_settings_display.jsx [hunk 8]
```diff
@@ -54,12 +54,6 @@ export default class UserSettingsDisplay extends React.Component {
         currentUserTimezone: PropTypes.string,
         enableTimezone: PropTypes.bool,
         shouldAutoUpdateTimezone: PropTypes.bool,
-        militaryTime: PropTypes.string,
-        teammateNameDisplay: PropTypes.string,
-        channelDisplayMode: PropTypes.string,
-        messageDisplay: PropTypes.string,
-        collapseDisplay: PropTypes.string,
-        linkPreviewDisplay: PropTypes.string,
```
Comment (mgdelacroix) [inline components/user_settings/display/user_settings_display.jsx]: All these props are in use in the function `getDisplayStateFromProps`

File: components/youtube_video/youtube_video.jsx [hunk 9]
```diff
@@ -15,11 +15,8 @@ const ytRegex = /(?:http|https):\/\/(?:www\.|m\.)?(?:(?:youtube\.com\/(?:(?:v\/)
 
 export default class YoutubeVideo extends React.PureComponent {
     static propTypes = {
-        channelId: PropTypes.string.isRequired,
-        currentChannelId: PropTypes.string.isRequired,
         hasImageProxy: PropTypes.bool.isRequired,
         link: PropTypes.string.isRequired,
-        show: PropTypes.bool.isRequired,
```
Comment (mgdelacroix) [inline components/youtube_video/youtube_video.jsx]: These three props are in use in (at least) the function `updateStateFromProps`

File: plugins/channel_header_plug/channel_header_plug.jsx [hunk 10]
```diff
@@ -88,11 +88,6 @@ export default class ChannelHeaderPlug extends React.PureComponent {
 
         channel: PropTypes.object.isRequired,
         channelMember: PropTypes.object.isRequired,
-
-        /*
-         * Logged in user's theme
-         */
-        theme: PropTypes.object.isRequired,
```
Comment (mgdelacroix) [inline plugins/channel_header_plug/channel_header_plug.jsx]: We can stop passing `theme` from the index

File: plugins/mobile_channel_header_plug/mobile_channel_header_plug.jsx [hunk 11]
```diff
@@ -19,11 +19,6 @@ export default class MobileChannelHeaderPlug extends React.PureComponent {
 
         channel: PropTypes.object.isRequired,
         channelMember: PropTypes.object.isRequired,
-
-        /*
-         * Logged in user's theme
-         */
-        theme: PropTypes.object.isRequired,
```
Comment (mgdelacroix) [inline plugins/mobile_channel_header_plug/mobile_channel_header_plug.jsx]: We can stop passing `theme` from the index

**Comment by skiplist** [comment]
@mgdelacroix Probably was more work for you to comment on this PR than for me to do it :)

**Comment by skiplist** [comment]
I don't like that ESLint is incorrectly identifying props as unused.

**Comment by mgdelacroix** [comment]
@skiplist yeah, it was a bummer when I found those almost at the end of the PR

**Pull request closed without merging**